src/js/base/js-numbers.js

```diff
@@ -1201,16 +1201,44 @@ define("pyret-base/js/js-numbers", function() {
       return bnRemainder.call(m, n);
     });
 
+  var splitIntintoMantissaExpt = function(s) {
+    var str = s.toString();
+    var c0 = str[0];
+    var sign = '';
+
+
+    if (c0 === '-' || c0 === '+') {
+      str = str.substring(1);
+      if (c0 === '-') {
+        sign = '-';
+      }
+    }
+
+    var d0 = str[0];
+    str = str.substring(1);
+
+    var mantissa = Number(sign + d0 + '.' + str);
```

Comment:
Could definitely do with a comment or two here; I've read this code three times now and I'm not quite sure whether it works or not...  Is the invariant here that `s` is a (big) integer, so there should be no decimal point in the `toString` output, that you are then inserting?  Is the expt off by one below?